lib/charms/opensearch/v0/opensearch_users.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""OpenSearch user helper functions.
+
+These functions wrap around some API calls used for user management.
+"""
+
+import logging
+from typing import Any, Dict, List
+
+from charms.opensearch.v0.opensearch_distro import OpenSearchDistribution
+
+logger = logging.getLogger(__name__)
+
+
+USER_ENDPOINT = "/_plugins/_security/api/internalusers"
+ROLE_ENDPOINT = "/_plugins/_security/api/roles"
+
+
+class OpenSearchUserMgmtError(Exception):
+    """Base exception class for OpenSearch user management errors."""
+
+
+def create_role(
+    opensearch: OpenSearchDistribution,
+    role_name: str,
+    permissions: Dict = {},
```

Comment:
Giving a “mutable” object as a default value is dangerous in Python, as the said argument is only evaluated during the `def` phase. Though may not necessarily be the case here, it should still be avoided.

You could rather make it optional the args: `permissions: Optional[Dict[str, str]]`
then later in the code do: `{**(permissions or {}), **...}`